cmake/dimension.cmake

```diff
@@ -5,15 +5,9 @@
 set(DIMENSION_INFILE ${PDAL_SRC_DIR}/Dimension.json)
 set(DIMENSION_OUTFILE ${CMAKE_CURRENT_BINARY_DIR}/include/pdal/Dimension.hpp)
 
-# In cross compiling scenarios, the location of the native DIMBUILDER_EXECUTABLE
-# must be provided to the CMake invocation.
-IF (CMAKE_CROSSCOMPILING)
-    SET(DIMBUILDER "${DIMBUILDER_EXECUTABLE}")
-ELSE (CMAKE_CROSSCOMPILING)
-    SET(DIMBUILDER dimbuilder)
-ENDIF (CMAKE_CROSSCOMPILING)
-
+# In cross compiling scenarios, you might want to set CMAKE_CROSSCOMPILING_EMULATOR
```

Comment:
@hobu: You just did this, right? I don't know what CMAKE_CROSSCOMPILING_EMULATOR is even after reading the doc (no surprise, given this is cmake doc). Or @xantares, can you explain?

Reply:
I do know that my use of `CMAKE_CROSSCOMPILING` worked for me, and I had seen usage of it around. IDK if that's the prescribed way to do this though. Happy to have a patch by anyone that does a lot more cross compilation stuff.

Reply:
@abellgithub for example when I cross-compile from a linux host to windows target I just set CMAKE_CROSSCOMPILING_EMULATOR to the wine executable, then cmake automagically knows that it has to run the dimbuilder.exe rule with wine, that simplifies the cmake logic.

Reply:
@xantares: I'm not sure how this will work with general X-compiling, but I confess that I usually run from cmake magic.

@hobu: I know if this will address your conda M1 issue. Can you check before we commit this? I don't want to break what you just got going.

Reply:
`wine` may run from Linux magic. 

For general cross-build, I would like to know if I could use a host `dimbuilder`?

Reply:
The conda-forge cross build scenario is building on a osx-x86_64 host for osx-arm64. In that case, we need to build and execute `dimbuilder` on the host machine's architecture (x86_64), run that, and then continue compiling for arm64. We're not running in an emulator scenario, so I'm not sure if `CMAKE_CROSSCOMPILING_EMULATOR` applies.

Reply:
in that case we should give the choice to use either DIMBUILDDER_EXECUTABLE or CMAKE_CROSSCOMPILING_EMULATOR, which wasnt possible previously.

Reply:
in that case we could offer to choose either DIMBUILDER_EXECUTABLE or CMAKE_CROSSCOMPILING_EMULATOR:
use